homeassistant/components/monzo/__init__.py

```diff
@@ -33,6 +34,8 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     hass.data.setdefault(DOMAIN, {})[entry.entry_id] = coordinator
     await hass.config_entries.async_forward_entry_setups(entry, PLATFORMS)
 
+    await register_services(hass)
```

Comment:
Sorry, I'm confused, that call is already in `async_setup_entry`? I though `async_setup` was for setup without a config entry?

Reply:
Yep, exactly. And services should be setup without the config entry (if it aren't `entity_services`) and check for existence of the `config_entry` (if it is required) in the service, like you already do

Reply:
Are you sure this is correct? Only I can't find any mention in the docs and a quick regex search shows there are at least 3 times as many integrations which do it in `async_setup_entry` rather than `async_setup`.

Reply:
yes, but it changed only recently. Here is another example PR. This is the guideline "_Integration services should be registered in async_setup and not removed. They need to handle missing or not loaded config entry during the service call_." (taken from martinhjelmare on discord).

Reply:
https://github.com/home-assistant/core/pull/116459/files another PR where this was explicitly changed

Reply:
Ah, got it, thanks!